core/rawdb/accessors_state.go

```diff
@@ -27,6 +27,7 @@ import (
 // ReadPreimage retrieves a single preimage of the provided hash.
 func ReadPreimage(db ethdb.KeyValueReader, hash common.Hash) []byte {
 	data, _ := db.Get(preimageKey(hash))
+	preimageMissCounter.Inc(1)
```

Comment:
Hmm, shouldn't we update the hit/miss metrics respectively according to the data?

Reply:
If so, seems we need to move the metrics into the trieb package. 

Or we can rename those two metrics to preimage read/write counter.

Reply:
What I meant is:

if len(data) == 0 {
    // count miss
} else {
    // count hit
}

Reply:
Thanks for the explain, I thought it was cache hit ratio